mutation_fragment.cc

```diff
@@ -100,35 +100,44 @@ mutation_fragment::mutation_fragment(const schema& s, reader_permit permit, stat
     : _kind(kind::static_row), _data(std::make_unique<data>(std::move(permit)))
 {
     new (&_data->_static_row) static_row(std::move(r));
-    _data->_memory.reset(reader_resources::with_memory(calculate_memory_usage(s)));
+    reset_memory(s);
 }
 
 mutation_fragment::mutation_fragment(const schema& s, reader_permit permit, clustering_row&& r)
     : _kind(kind::clustering_row), _data(std::make_unique<data>(std::move(permit)))
 {
     new (&_data->_clustering_row) clustering_row(std::move(r));
-    _data->_memory.reset(reader_resources::with_memory(calculate_memory_usage(s)));
+    reset_memory(s);
 }
```

Comment:
I remember commending that this is unneeded because the destructor will do that for us.

Reply:
Yes, and I responded that the destructor is not called if the exception is thrown in the constructor.

Reply:
Right.